Description: update AP properties Flags,WPA,RSN 
Author: electricface <songwentai@deepin.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: 2019-11-07

--- network-manager-1.14.6.2.orig/src/devices/wifi/nm-wifi-ap.c
+++ network-manager-1.14.6.2/src/devices/wifi/nm-wifi-ap.c
@@ -796,6 +796,8 @@ nm_wifi_ap_update_from_properties (NMWif
 
 	if (g_variant_lookup (properties, "Privacy", "b", &b) && b)
 		changed |= nm_wifi_ap_set_flags (ap, priv->flags | NM_802_11_AP_FLAGS_PRIVACY);
+	else
+		changed |= nm_wifi_ap_set_flags (ap, priv->flags &~ NM_802_11_AP_FLAGS_PRIVACY);
 
 	v = g_variant_lookup_value (properties, "WPS", G_VARIANT_TYPE_VARDICT);
 	if (v) {
@@ -872,13 +874,13 @@ nm_wifi_ap_update_from_properties (NMWif
 
 	v = g_variant_lookup_value (properties, "WPA", G_VARIANT_TYPE_VARDICT);
 	if (v) {
-		changed |= nm_wifi_ap_set_wpa_flags (ap, priv->wpa_flags | security_from_vardict (v));
+		changed |= nm_wifi_ap_set_wpa_flags (ap, security_from_vardict (v));
 		g_variant_unref (v);
 	}
 
 	v = g_variant_lookup_value (properties, "RSN", G_VARIANT_TYPE_VARDICT);
 	if (v) {
-		changed |= nm_wifi_ap_set_rsn_flags (ap, priv->rsn_flags | security_from_vardict (v));
+		changed |= nm_wifi_ap_set_rsn_flags (ap, security_from_vardict (v));
 		g_variant_unref (v);
 	}
 
